plot_features.py

```
@@ -132,40 +78,66 @@ def plot_features_file(featurefile, mapInfo): #{{{
+            props['marker'] = marker
+
+        if bounds is None:
+            bounds = list(shape.bounds)
+        else:
+            # expand the bounding box
+            bounds[0] = min(bounds[0],shape.bounds[0])
+            bounds[1] = min(bounds[1],shape.bounds[1])
+            bounds[2] = max(bounds[2],shape.bounds[2])
+            bounds[3] = max(bounds[3],shape.bounds[3])
 
-        color_num = feature_num % len(colors)
-        marker_num = feature_num % len(markers)
+        for mapType in mapInfo:
+            (ax, projection, plotFileName, fig) = mapInfo[mapType]
 
-        try:
-            if polytype == 'MultiPolygon':
-                for poly in coords:
-                    for shape in poly:
-                        plot_poly(mapInfo,shape,colors[color_num])
-            elif polytype == 'Polygon' or polytype == 'MultiLineString':
-                for poly in coords:
-                    plot_poly(mapInfo,poly,colors[color_num])
-            elif polytype == 'LineString':
-                plot_poly(mapInfo,coords,colors[color_num],filled=False)
-            elif polytype == 'Point':
-                plot_point(mapInfo,coords,markers[marker_num],colors[color_num])
-            else:
-                assert False, 'Geometry %s not known.'%(polytype)
-        except:
-            print 'Error plotting %s'%(feature)
+            ax.add_geometries((shape,), crs=refProjection, **props)
 
         feature_num = feature_num + 1
 
+    box = shapely.geometry.box(bounds[0],bounds[1],bounds[2],bounds[3])
+
+
     for mapType in mapInfo:
         (ax, projection, plotFileName, fig) = mapInfo[mapType]
         print 'saving ' + plotFileName
+
+        boxProjected = projection.project_geometry(box, src_crs=refProjection)
+        try:
+            x1, y1, x2, y2 = boxProjected.bounds
+            ax.set_xlim([x1, x2])
+            ax.set_ylim([y1, y2])
+        except ValueError:
+            # something wrong with box projection?
```

@pwolfram, what you say makes sense.  I'll just remove the try/except here.  I figured if the projection failed it wasn't a big deal to default back to a global projection but I agree that there should be a warning at a minimum.

I will double check but I think the reason I was getting value errors was a bug in the code.  So if I don't get the conditions where boxProjected.bounds is empty with any cases I try, I'll just assume it only happens when something is going wrong and it should raise an exception.